t/add-del.t

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+# a TAP compatible test
+
+# quick and dirty: fail early
+set -eu
+
+# test plan
+echo 1..8
+
+# usage:
+#  foo || tapfail
+#  echo "ok"
+# this will tell TAP "not ok" IFF foo has failed
+tapfail() {
+  echo -n "not "
+}
+
+YAST=/usr/sbin/yast
+
+echo "# assume and check that eth0 exists, as a base for the vlan"
+ip addr show dev eth0 || tapfail
+echo "ok 1 eth0: exists already"
+
+echo "# add a (virtual) interface"
+$YAST lan add name=vlan50 ethdevice=eth0 bootproto=dhcp
+echo "ok 2 vlan50: added"
+
+# check it has worked
+ip addr show dev vlan50 || tapfail
+echo "ok 3 vlan50: interface exists"
+
+ls /etc/sysconfig/network/ifcfg-vlan50 || tapfail
+echo "ok 4 vlan50: ifcfg exists"
+
+echo "# find it"
+# the embarassing part: no way to identify by "vlan50"
+ID=$($YAST lan list |& grep Virtual.LAN | cut -f1)
+echo "ok 5 vlan50: yast specific ID found"
+
+echo "# delete it"
+$YAST lan delete id=$ID verbose
```

Comment:
Good question. I think we are using tapfail in a quite arbitrary way. How should we use it in my opinion?

`prove` interrupts the execution of the whole script if some line returns non-zero. So in my opinion we should never use tapfail when the failure is critical and would break the following steps. So IMO, 1 and 2 should not use tapfail because if they fail we really want to abort (it will be reported as an error by prove even without the `echo not`.

The rest of the tests should use tapfail because we can still continue trying after them.